### PR TITLE
Support secondary indexes on DynamoDB

### PR DESCRIPTION
**Pull request by yito88**
https://scalar-labs.atlassian.net/browse/DLT-7881

- Use global secondary indexes of DynamoDB
  - The consistency of global secondary indexes on Dynamo DB is eventual consistency
  - https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/GSI.html

### CODE REVIEW

**Review by feeblefakie — APPROVED** [review]
LGTM!

File: src/integration-test/java/com/scalar/db/storage/dynamo/DynamoIntegrationTest.java [hunk 1]
```diff
@@ -154,6 +156,50 @@ public void get_GetWithProjectionsGiven_ShouldRetrieveSpecifiedValues()
     assertThat(actual.get().getValue(COL_NAME5).isPresent()).isFalse();
   }
 
+  @Test
+  public void get_GetGivenForIndexedColumn_ShouldGet() throws ExecutionException {
```
Comment (brfrn169) [inline src/integration-test/java/com/scalar/db/storage/dynamo/DynamoIntegrationTest.java:160]: This is just a comment. We have the same integration code for each storage implementation. I'm wondering if we can unify the same integration test code for all storage implementations.

**Review by brfrn169 — APPROVED** [review]
LGTM!

File: src/integration-test/java/com/scalar/db/storage/dynamo/DynamoIntegrationTest.java (hunk 1 above)
Comment (feeblefakie) [reply src/integration-test/java/com/scalar/db/storage/dynamo/DynamoIntegrationTest.java:160]: @brfrn169 Let's do it! I'll create a story for that.

**Pull request merged**